win32/service.py

```diff
@@ -0,0 +1,289 @@
+#!/opt/datadog-agent/embedded/bin/python
+"""
+    Datadog
+    www.datadoghq.com
+    ----
+    Cloud-Scale Monitoring. Monitoring that tracks your dynamic infrastructure.
+
+    Licensed under Simplified BSD License (see LICENSE)
+    (C) Datadog, Inc. 2010-2016 all rights reserved
+"""
+
+# set up logging before importing any other components
+from config import initialize_logging  # noqa
+initialize_logging('service')
+
+# stdlib
+from collections import deque
+import logging
+import multiprocessing
+import os
+import psutil
+import time
+
+# win32
+import win32serviceutil
+import servicemanager
+import win32service
+
+# project
+from config import get_config, get_config_path, get_confd_path
+from jmxfetch import JMXFetch
+from utils.jmx import JMXFiles
+from utils.windows_configuration import get_registry_conf, update_conf_file
+
+
+log = logging.getLogger('service')
+
+
+SERVICE_SLEEP_INTERVAL = 1
+
+
+class AgentSvc(win32serviceutil.ServiceFramework):
+    _svc_name_ = "DatadogAgent"
+    _svc_display_name_ = "Datadog Agent"
+    _svc_description_ = "Sends metrics to Datadog"
+    _MAX_JMXFETCH_RESTARTS = 3
+    devnull = None
+
+    def __init__(self, args):
+        win32serviceutil.ServiceFramework.__init__(self, args)
+
+        AgentSvc.devnull = open(os.devnull, 'w')
+
+        config = get_config(parse_args=False, can_query_registry=False)
+        if config['api_key'] == 'APIKEYHERE':
+            self._update_config_file(config)
+
+        # Let's have an uptime counter
+        self.start_ts = None
+
+        # find the main agent dir, for instance C:\Program Files\Datadog\Datadog Agent\agent
```

Comment:
the comment is incorrect right? Think this should read `C:\Program Files\Datadog\Datadog Agent`